src/core/Akka/IO/TcpOutgoingConnection.cs

```diff
@@ -24,7 +25,7 @@ internal sealed class TcpOutgoingConnection : TcpConnection
         private readonly Tcp.Connect _connect;
         
         public TcpOutgoingConnection(TcpExt tcp, IActorRef commander, Tcp.Connect connect)
-            : base(tcp, new Socket(connect.RemoteAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp) { Blocking = false }, connect.PullMode)
```

Comment:
This was the change that caused the most issues with DNS. Passing in a `DnsEndpoint` with an unspecified `AddressFamily` caused DNS to not work at all, which is how most people use it.  We never specified the `AddressFamily` in Akka.IO prior to 1.3.0 except on Mono to work around an issue where binding to an IPV6 socket was unsupported. That issue appears to be fixed in the more recent versions of Mono, although as I noted in this PR there appears to be issues at the OS level with DNS resolution and IPV6 still.